src/models/chan.js

```diff
@@ -136,11 +136,29 @@ Chan.prototype.removeUser = function(user) {
 	this.users.delete(user.nick.toLowerCase());
 };
 
-Chan.prototype.toJSON = function() {
-	var clone = _.clone(this);
-	clone.users = []; // Do not send user list, the client will explicitly request it when needed
-	clone.messages = clone.messages.slice(-100);
-	return clone;
+Chan.prototype.getFilteredClone = function(lastActiveChannel, lastMessage) {
+	// Perform manual cloning of channel object for better control of performance and memory usage
+	return Object.keys(this).reduce((newChannel, prop) => {
+		if (prop === "users") {
```

Comment:
Does this need a `if <...>.hasOwnProperty` or whatnot?
Also, any specific reason to use this way instead of `for ... in`? I'd expect `for ... in` to be sliiightly faster, but you most likely know better.

Reply:
`for in` should work.

Reply:
I could have sworn that `for ... in` + `hasOwnProperty` + whatever necessary would have been faster than `Object.keys(...).reduce` (because we loop only once).